auth/src/main/java/com/firebase/ui/auth/util/SmartLock.java

```diff
@@ -133,77 +140,100 @@ public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
                 }
             } else {
-                finish(RESULT_FIRST_USER, getIntent());
+                mActivity.finish(RESULT_OK, mActivity.getIntent());
             }
         }
     }
 
     @Override
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
+
         if (requestCode == RC_SAVE) {
             if (resultCode == RESULT_OK) {
                 if (BuildConfig.DEBUG) {
                     Log.d(TAG, "SAVE: OK");
                 }
-                finish(RESULT_OK, getIntent());
+                mActivity.finish(RESULT_OK, new Intent());
             } else {
                 Log.e(TAG, "SAVE: Canceled by user");
-                finish(RESULT_FIRST_USER, getIntent());
+                mActivity.finish(RESULT_OK, new Intent());
             }
         } else if (requestCode == RC_UPDATE_SERVICE) {
             if (resultCode == RESULT_OK) {
-                Credential credential = new Credential.Builder(mEmail).setPassword(mPassword).build();
-                mActivityHelper.getCredentialsApi()
+                Credential credential = new Credential.Builder(mEmail).setPassword(mPassword)
+                        .build();
+                Auth.CredentialsApi
                         .save(mCredentialsApiClient, credential)
                         .setResultCallback(this);
             } else {
                 Log.e(TAG, "SAVE: Canceled by user");
-                finish(RESULT_FIRST_USER, getIntent());
+                mActivity.finish(RESULT_OK, new Intent());
             }
         }
     }
 
-    public static Intent createIntent(
-            Context context,
-            FlowParameters flowParams,
-            FirebaseUser user,
-            @Nullable String password,
-            @Nullable String provider) {
-
-        String photoUrl = user.getPhotoUrl() != null ? user.getPhotoUrl().toString() : null;
-        return ActivityHelper.createBaseIntent(context, SaveCredentialsActivity.class, flowParams)
-                .putExtra(ExtraConstants.EXTRA_NAME, user.getDisplayName())
-                .putExtra(ExtraConstants.EXTRA_EMAIL, user.getEmail())
-                .putExtra(ExtraConstants.EXTRA_PASSWORD, password)
-                .putExtra(ExtraConstants.EXTRA_PROVIDER, provider)
-                .putExtra(ExtraConstants.EXTRA_PROFILE_PICTURE_URI, photoUrl);
+    /**
+     * If SmartLock is enabled and Google Play Services is available, save the credentials.
+     * Otherwise, finish the calling Activity with RESULT_OK.
+     *
+     * @param activity     the calling Activity.
+     * @param parameters   calling Activity flow parameters.
+     * @param firebaseUser Firebase user to save in Credential.
+     * @param password     (optional) password for email credential.
+     * @param provider     (optional) provider string for provider credential.
+     */
+    public static SmartLock getInstance(AppCompatBase activity,
```

Comment:
Ohhhhh. I think I'm getting it now. You can pass in things like Activities to a fragment because they are a more conventional Java object: you can just do method injection.